app/experimenter/experiments/models/nimbus.py

```diff
@@ -531,6 +533,37 @@ def __str__(self):
         return self.name
 
 
+# Helper to ensure branch screenshot filenames have controlled unique paths
+def nimbus_branch_screenshot_upload_to(screenshot, filename):
+    branch_slug = screenshot.branch.slug
+    experiment_slug = screenshot.branch.experiment.slug
+    ext = filename.split('.')[-1].lower()
+    filename = "%s.%s" % (uuid.uuid4(), ext)
+    return f"{experiment_slug}/{branch_slug}/{filename}"
```

Comment:
I remembered I did this back on MDN. Turns out it's easy to try uploading `Capture.PNG` multiple times for different screenshots, so using user-supplied filenames is trouble. So, this tries to maintain unique filenames for every upload, associated with the owning model.

Reply:
Yeah not a bad idea.  I'd suggest using `os.path.join` instead of string formatting though.